apps/qubit/modules/actor/templates/browseSuccess.php

```diff
@@ -102,25 +102,11 @@
       </span>
     <?php endif; ?>
 
-    <?php if (isset($repository)): ?>
-      <?php echo get_component('search', 'filter', array('object' => $repository, 'param' => 'repository')) ?>
-    <?php endif; ?>
-
-    <?php if (isset($entityType)): ?>
-      <?php echo get_component('search', 'filter', array('object' => $entityType, 'param' => 'entityType')) ?>
-    <?php endif; ?>
-
-    <?php if (isset($occupation)): ?>
-      <?php echo get_component('search', 'filter', array('object' => $occupation, 'param' => 'occupation')) ?>
-    <?php endif; ?>
-
-    <?php if (isset($places)): ?>
-      <?php echo get_component('search', 'filter', array('object' => $places, 'param' => 'places')) ?>
-    <?php endif; ?>
-
-    <?php if (isset($subjects)): ?>
-      <?php echo get_component('search', 'filter', array('object' => $subjects, 'param' => 'subjects')) ?>
-    <?php endif; ?>
+      <?php echo get_component('search', 'filter', array('object' => @$repository, 'param' => 'repository')) ?>
+      <?php echo get_component('search', 'filter', array('object' => @$entityType, 'param' => 'entityType')) ?>
+      <?php echo get_component('search', 'filter', array('object' => @$occupation, 'param' => 'occupation')) ?>
+      <?php echo get_component('search', 'filter', array('object' => @$places, 'param' => 'places')) ?>
+      <?php echo get_component('search', 'filter', array('object' => @$subjects, 'param' => 'subjects')) ?>
```

Comment:
I see now why you checked the vars before calling the component and I'm having conflicts about this. I don't like the `@` as it may hide further issues, but in this cases it will only hide a notice. We show many notices like those in debug mode but I like the idea of avoiding as many as possible. I don't like to introduce that the operator but I think it's excessive to initiate all those vars as `null` in the action.

Reply:
Thinking a little more about this, we could put all the filter objects into a `param => object` array and loop over it to add the components, similar to the hidden fields. That way we could avoid the `@` operator and it will also simplify a little the template.

Would that be possible? If you're running out of time, we could add a note about this to the issue you created to use the component in the IO browse page. Please, add me as a watcher to that issue ;)

Thanks!

Reply:
Hi @radda... yeah, having a param => object array was what I was thinking too. Given we're tight on time for this I'll add note to the issue about the other filter refactoring. Thanks again for quickly reviewing this!